### PR TITLE
tests: use snapcraft and lxd with latest fixes where possible

### PR DESCRIPTION
**Pull request by ernestl**
Use latest (edge) versions of snapcraft and lxd to avoid bugs and ensure compatibility. The LXD bug causes this issue in a few tests: https://paste.ubuntu.com/p/y3RGD2hjb5/, and the snapcraft bug cannot deal with the new LXD version naming that may include "LTS".

lxd bug: https://github.com/canonical/lxd-pkg-snap/pull/379
snapcraft bug: https://chat.canonical.com/canonical/pl/6w13o16npbffup3obepqiig3tr

### CODE REVIEW

**Review by zyga — APPROVED** [review]
The eternal dance of version continues :)

I recall us doing this so many times before, that it's now tradition near the release ;)

File: tests/nested/manual/devmode-snaps-can-run-other-snaps/task.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ environment:
   # build the snap with lxd
   SNAPCRAFT_BUILD_ENVIRONMENT: lxd
 
+  # pin LXD to 5.20/stable to ensure to prevent snapcraft problem with handling new version naming that may include "LTS"
+  LXD_SNAP_CHANNEL: 5.20/stable
```
[the remark this answers is not in the document]
Comment (Meulengracht) [reply tests/nested/manual/devmode-snaps-can-run-other-snaps/task.yaml:25]: Yes - this will override the spread.yaml defined key AFAIK

**Review by Meulengracht — APPROVED** [review]
lgtm, however static checks are failing

**Comment by ernestl** [comment]
> lgtm, however static checks are failing

Yes, details was missing, added this now.

**Pull request merged**